libraries/src/Form/Field/CheckboxField.php

```diff
@@ -137,7 +137,7 @@ public function setup(\SimpleXMLElement $element, $value, $group = null)
     protected function getLayoutData()
     {
         $data            = parent::getLayoutData();
-        $data['value']   = $this->default ?: '1';
+        $data['value']   = $this->default || $this->default === '0' ? $this->default : '1';
```

Comment:
I would recommend the following changes

```suggestion
        $data['value']   = strlen($this->default) ? $this->default : '1';
```

Reply:
@bembelimen   Your method is not suitable, since `$this->default` may be undefined and have a `NULL` value.

Reply:
Both is wrong :)
This should be okau:
```php
$data['value'] = $this->default !== null && $this->default !== '' ? $this->default : '1';
```